### PR TITLE
Improve Build disables

### PR DESCRIPTION
**Pull request by amilcarlucas**

### CODE REVIEW

File: ArduCopter/APM_Config.h [hunk 1]
```diff
@@ -21,7 +21,7 @@
 //#define FRSKY_TELEM_ENABLED   DISABLED            // disable FRSky telemetry
 //#define ADSB_ENABLED          DISABLED            // disable ADSB support
 //#define PRECISION_LANDING     DISABLED            // disable precision landing using companion computer or IRLock sensor
-//#define BEACON_ENABLED        DISABLED            // disable beacon support
+//#define BEACON_ENABLED        DISABLED            // disable localization beacon support
```
Comment (OXINARF) [inline ArduCopter/APM_Config.h]: I actually prefer the older comment, we never refer to beacon support/library as localization beacon.

**Comment by amilcarlucas** [comment]
done

File: ArduCopter/ArduCopter.cpp [hunk 2]
```diff
@@ -120,7 +124,9 @@ const AP_Scheduler::Task Copter::scheduler_tasks[] = {
 #if FRAME_CONFIG == HELI_FRAME
     SCHED_TASK(check_dynamic_flight,  50,     75),
 #endif
+#if LOGGING_ENABLED == ENABLED
     SCHED_TASK(fourhundred_hz_logging,400,    50),
```
Comment (peterbarker) [inline ArduCopter/ArduCopter.cpp]: Maybe move this down to the other logging calls?

Comment (amilcarlucas) [reply ArduCopter/ArduCopter.cpp]: I avoided do that functional change, but if you say it is OK, sure, I can do that.

**Comment by peterbarker** [comment]
On Thu, 1 Mar 2018, Amilcar Lucas wrote:

> > @@ -120,7 +124,9 @@ const AP_Scheduler::Task Copter::scheduler_tasks[] = {
>  #if FRAME_CONFIG == HELI_FRAME
>      SCHED_TASK(check_dynamic_flight,  50,     75),
>  #endif
> +#if LOGGING_ENABLED == ENABLED
>      SCHED_TASK(fourhundred_hz_logging,400,    50),
> 
> I avoided do that functional change, but if you say it is OK, sure, I can do
> that.

Hmmm.  Well, most of the time flying is more important than logging :-)

But you're right, you should keep this PR NFC.

**Comment by OXINARF** [comment]
More squash, slightly changed commit messages and merged, thanks!

**Pull request closed without merging**